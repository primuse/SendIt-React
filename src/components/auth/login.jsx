import React, { Component } from "react";
import { LoginForm, SignupForm } from "../forms/loginForm.jsx";
import { Link } from "react-router-dom";
import "../../css/modules.css";
import "../../css/style.css";
import "../../css/login.css";

export const Login = () => {
  return (
    <section id="login">
      <header>
        <ul>
          <li id="head-brand">
            <Link to="/">
              Send<span className="head-span">IT</span>
            </Link>
          </li>
        </ul>
        <ul className="ml-auto mt-25">
          <li>
            <Link to="/signup">Signup</Link>
          </li>
        </ul>
      </header>
      <div className="banner">
        <div className="sendit-text mb-34">
          <h5>
            The <span className="head-span">Courier</span> service app you've
            been waiting for
          </h5>
          <p>SendIT makes reaching the world easier!</p>
        </div>
        <LoginForm />
      </div>
    </section>
  );
};

export const Signup = () => {
  return (
    <section id="login">
      <header>
        <ul>
          <li id="head-brand">
            <Link to="/">
              Send<span class="head-span">IT</span>
            </Link>
          </li>
        </ul>
        <ul class="ml-auto mt-25">
          <li>
            <Link to="/">Login</Link>
          </li>
        </ul>
      </header>
      <div class="banner">
        <div class="sendit-text">
          <h5>
            The <span class="head-span">Courier</span> service app you've been
            waiting for
          </h5>
          <p class="mb-34">
            Register to experience our world class courier services
          </p>
        </div>
        <div id="form-box">
          <div class="text-center">
            <SignupForm />
          </div>
        </div>
      </div>
    </section>
  );
};
