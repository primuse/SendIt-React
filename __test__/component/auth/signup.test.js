import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import { MemoryRouter } from 'react-router-dom';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import thunk from 'redux-thunk';
import { Signup } from '../../../src/components/auth/login.jsx';
import LoginForm from '../../../src/components/forms/loginForm.jsx';

const initialState = {
  auth: {
    isLoading: false
  }
};
const mockStore = configureStore([thunk]);
const store = mockStore(initialState);


describe('Login page component', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<MemoryRouter>
        <Provider store={store}>
          <Signup />
        </Provider>
      </MemoryRouter>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
