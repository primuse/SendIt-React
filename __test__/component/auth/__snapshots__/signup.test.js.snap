// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Signup page component renders correctly 1`] = `
<section
  id="login"
>
  <header>
    <ul>
      <li
        id="head-brand"
      >
        <a
          href="/"
          onClick={[Function]}
        >
          Send
          <span
            className="head-span"
          >
            IT!
          </span>
        </a>
      </li>
    </ul>
    <ul
      className="ml-auto mt-25"
    >
      <li>
        <a
          href="/"
          onClick={[Function]}
        >
          Login
        </a>
      </li>
    </ul>
  </header>
  <div
    className="banner"
  >
    <div
      className="sendit-text"
    >
      <h5>
        The 
        <span
          className="head-span"
        >
          Courier
        </span>
         service app you've been waiting for
      </h5>
      <p
        className="mb-34"
      >
        Register to experience our world class courier services
      </p>
    </div>
    <div
      id="form-box"
    >
      <div
        className="text-center"
      >
        <form
          id="signupForm"
          onSubmit={[Function]}
        >
          <div
            className="column is-two form-row"
          >
            <input
              id="firstName"
              name="firstName"
              onChange={[Function]}
              placeholder="First name"
              required={true}
              type="text"
              value=""
            />
            <input
              id="lastName"
              name="lastName"
              onChange={[Function]}
              placeholder="Last name"
              required={true}
              type="text"
              value=""
            />
          </div>
          <div
            className="column is-two form-row"
          >
            <input
              id="email"
              name="email"
              onChange={[Function]}
              placeholder="Email"
              required={true}
              type="email"
              value=""
            />
            <input
              id="password"
              name="password"
              onChange={[Function]}
              placeholder="Password"
              required={true}
              type="password"
              value=""
            />
          </div>
          <button
            className="btn lg is-outlined"
            type="submit"
          >
            Signup
          </button>
        </form>
      </div>
    </div>
  </div>
</section>
`;
